src/System.Runtime.Extensions/src/System/Environment.cs

```diff
@@ -65,35 +65,12 @@ public static string CommandLine
         {
             get
             {
-                StringBuilder sb = StringBuilderCache.Acquire();
-
-                foreach (string arg in GetCommandLineArgs())
-                {
-                    bool containsQuotes = false, containsWhitespace = false;
-                    foreach (char c in arg)
-                    {
-                        if (char.IsWhiteSpace(c))
-                        {
-                            containsWhitespace = true;
-                        }
-                        else if (c == '"')
-                        {
-                            containsQuotes = true;
-                        }
-                    }
-
-                    string quote = containsWhitespace ? "\"" : "";
-                    string formattedArg = containsQuotes && containsWhitespace ? arg.Replace("\"", "\\\"") : arg;
-
-                    sb.Append(quote).Append(formattedArg).Append(quote).Append(' ');
-                }
-
-                if (sb.Length > 0)
-                {
-                    sb.Length--;
-                }
-
-                return StringBuilderCache.GetStringAndRelease(sb);
+                var args = GetCommandLineArgs();
+                // Not to throw for quotes inside argv0. Since the input was constructed by the framework itself from the
+                // passed in command line, this shouldn't ever happen but if it ever does, throwing is a bit extreme here.
```

Comment:
I think we should rather change `PasteArguments.Paste` to not throw, at least on Linux -- quotes is valid directory/file name character on Linux.
IMO we should preserve the quotes by escaping them (double quotes) inside argv0.

@AtsushiKan what do you think?

Reply:
I think a way to go is to have `PasteArguments.Paste` ignore the `pasteFirstArgumentUsingArgV0Rules` parameter on Unix builds. This has the effect you want on Unix (no exception, emits an escaped quote.)

This business of having special rules for argv0 is for Windows compat and doesn't need to be exported to other OS's.